### ISSUE TITLE
Problems with interpolation in "ip_configuration" variable

### ISSUE THREAD

**Issue by hypnoglow** [GoogleCloudPlatform/terraform-google-sql-db#30]
I'm trying to deploy private postgresql using the module.

My module definition looks like this:
```hcl
module "postgres_foo" {
  source  = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
  version = "1.1.1"

  name       = "postgres-foo-${random_id.name.hex}"
  project_id = "${var.project_id}"
  region     = "${var.region}"
  zone       = "${var.zone_suffix}"
  database_version = "POSTGRES_9_6"

  ip_configuration = {
    ipv4_enabled = false
    private_network = "${google_compute_network.main.self_link}"
  }
}
```

Planning this results in error:

```
Error: module.postgres_foo.google_sql_database_instance.default: settings.0.ip_configuration: should be a list
```

However, if I do `terraform state show google_compute_network.main` and copy value of `self_link` directly, `terraform plan` works fine:

```hcl
module "postgres_foo" {
  source  = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
  version = "1.1.1"

  name       = "postgres-foo-${random_id.name.hex}"
  project_id = "${var.project_id}"
  region     = "${var.region}"
  zone       = "${var.zone_suffix}"
  database_version = "POSTGRES_9_6"

  ip_configuration = {
    ipv4_enabled = false
    private_network = "https://www.googleapis.com/compute/v1/projects/my-project/global/networks/main"
  }
}
```

I suppose it somehow relates to interpolation issues in module variables of complex types (e.g. maps), but I'm not sure. The only related issue I've found is this: https://github.com/hashicorp/terraform/issues/7241

**Comment by namusyaka** [comment]
@hypnoglow Thanks for the report.
Could you try to use list type for `ip_configuration` like [our example](https://github.com/GoogleCloudPlatform/terraform-google-sql-db/blob/02a3a5d1514cd082ab1afc3af6f19192cadeb76c/examples/mysql-and-postgres/main.tf#L93-L98)?

Of course, it's not smart, but would be worked as expected.

**Comment by thefirstofthe300** [comment]
@namusyaka I'm hitting this same issue and the issue doesn't appear to be related to not passing in a list value since the TF I'm using is
```
module "cloud_sql_mysql" {
  source  = "GoogleCloudPlatform/sql-db/google//modules/mysql"
  version = "1.1.1"

  project_id       = "${var.project_id}"
  name             = "${var.mysql_name}-${random_id.name.hex}"
  database_version = "${var.mysql_version}"
  region           = "${var.region}"
  zone             = "${var.zone}"
  tier             = "${var.tier}"

  ip_configuration = [{
    ipv4_enabled    = "false"
    require_ssl     = "true"
    private_network = "${var.private_network}"
  }]

  disk_size                       = "${var.disk_size}"
  maintenance_window_day          = "${var.maintenance_window_day}"
  maintenance_window_hour         = "${var.maintenance_window_hour}"
  maintenance_window_update_track = "${var.maintenance_window_update_track}"
  additional_users                = "${var.additional_users}"
}
```

**Comment by thefirstofthe300** [comment]
Found the bug. Terraform 0.11's terrible support for nested lists/maps strikes again.

PR incoming.

**Comment by namusyaka** [comment]
@thefirstofthe300 That's nice. #38 has been reviewed.

**Comment by namusyaka** [comment]
closed via #38
LMK if you still have any issues.
